### ISSUE TITLE
[MU4 Issue] Pressing tab while changing instrument names doesn't do anything

### ISSUE THREAD

**Issue by Grim-O** [musescore/MuseScore#14343]
When editing instrument names in the instruments tab, there's no option to tab to the next input box. It would be nice and would make the flow of changing several names (as if often the case because name on main score, abbreviated name and part name often stay similar) smoother by adding the ability to press tab to go to the next input box.

**Comment by skdonnelly** [comment]
This is one of those unfortunate situations where someone decided to use arrow keys instead of Tab, for no logical reason. So you can get to the next text field using Down Arrow. MuseScore has made it a game where in any given UI element you have to guess whether Tab, down arrow, or some other key will get you to the next field, ensuring that you always have to stop to think and guess rather than just using the standard that every app has used for decades. :( And to navigate through this panel, the Project properties dialog etc., you have to constantly switch between Tab and arrow keys instead of being able to keep your hand on the keyboard, the whole reason for many of us to use keyboard shortcuts in the first place. 

Also hitting Enter after changing an instrument name should close the dialog so that you can move to the next instrument. Currently you have to hit Esc, which normally would mean you do NOT want to save your changes, but in this case it does save your changes. There should be a cancel button, or Esc should cancel, not apply your changes.

**Comment by MarcSabatella** [comment]
The changes weren't made for "no logical reason" - they are in response to new developments in the accessibility field where there is a move toward more hierarchical organization of UI controls.  One can quibble about specific choices made in specific windows - that's fair enough - but condescending remarks that ignore or denigrate the importance of following newly developing accessibility standards are not helpful.

And yes, newly developing standards do mean that things may work differently than how they did in previous decades.

**Comment by skdonnelly** [comment]
Sorry if I came across that way, I just work in accessibility and usability and have never encountered this standard, and any references to Web, Windows and MACOS standards on the subject refer to tab order, being able to tab through UI elements etc. I would appreciate some links or references to learn more. As mentioned in various threads on this subject, "tab" is the standard, and trying to force a new standard on users, especially when it has been implemented in a very haphazard way with no visual cues, seems counterproductive. I honestly think developers are misinterpreting the use of the arrow keys to move within logically and visually grouped elements (file names in a list, items in a dropdown list, radio buttons that are together, buttons in a toolbar, etc.) vs. disparate elements such as form fields and buttons, which one would expect to be able to Tab to. But as mentioned I'm happy to read up if something new is emerging. I'm all for enhancing usability and accessibility as much as possible, but we must avoid "UX theatre" and pretending we know what users want or what is "best" for users, and rely on usability testing and feedback from real users to make these decisions. All the feedback I have seen is that this change is counterintuitive, but obviously I do not have access to internal feedback that may show otherwise. Is there any documentation of this feedback or reasons for this change? I searched through Github but didn't find much. Like the name change of the "Instruments" panel to the "Layout" panel, v3 worked fine in this respect, the project properties dialog allows you to tab to all text fields on the "Enter score information:" dialog, tab to all buttons and boxes on the "Choose template file" dialog, and use the arrow keys while in grouped items such as the list of instruments, or when you're on a row of buttons that are all together like the Back/Next/Finish/Cancel buttons. I don't think anyone is opposed to this, nor have I seen any comments to that effect. But when you can't tab through text fields on a basic dialog box, or have to use random arrow keys to move around, or have to guess at any given point whether you need to use Tab or arrow keys, there is obviously (to many of us) a problem.

**Comment by MarcSabatella** [comment]
The references I have seen have all been in-person discussions with blind users or accessibility-related mailing lists etc - I haven't seen any documentation myself.  But the general consensus has been, any window with many controls that are divided into multiple panes is a good candidate for hierarchical navigation, whereas simple dialog boxes are not.  The control at hand here is neither of these, so it isn't totally clear what the best practice might be for applications that use hierarchical navigation.  It's definitely a valid discussion to be had, since it is *not* a black & white no-brainer in either direction.  I just don't have any special insight here personally.

**Comment by jmwmusic** [comment]
The hierarchical model makes some sense for larger section navigation.  However, it is never used in dialogue boxes and many of the MuseScore panels once in them function generally more like permanently open dialogue boxes.  all comments on the MuseScore Accessibility list have been in favour of all actionable items being tabable and standard functioning regarding combo boxes, radio buttons, etc.  MuseScore is very non standard and I think it is a fair statement that the real world users don't like this aspect as it has come up in many discussions and surveys of blind users as something they would like to see changed.  The ability for speed to keep your hands on the keyboard rather than function keys and arrows is a valid point.  But I also think that any sighted user who is used to filling in forms will automatically hit Tab not arrows to move to the next item quickly, this is the norm in anything behaving as a form.

**Comment by skdonnelly** [comment]
Agree. Tab for fields and buttons, arrow key for grouped items such as radio buttons, dropdown lists, and buttons within a toolbar is the norm. Absent some evidence or testing indicating this new paradigm is helpful, I'd suggest sticking with the standard instead of trying to reinvent the wheel. Until an authority such as WebAIM or WCAG suggests this type of navigation, I wouldn't assume it is recommended or adopted widely. WebAIM keyboard navigation recommendations have specific recommendations for when to use arrow keys: https://webaim.org/techniques/keyboard/
WCAG doesn't have any specifics but web browsers, Windows etc. support Tab for fields and arrow keys for things like grouped controls per WebAIM. As more applications move towards being web-based, consistency with web browser controls is also logical to reduce cognitive load on users, they shouldn't have to remember which keys do what in which application. Per WCAG: "If content does not follow the platform/user agent conventions it may be difficult to use, as users will need to learn different interaction methods. As a best practice, content should follow the platform/user agent conventions." https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html

Ultimately, this implementation has resulted in all kinds of issues anyway, like having to use random arrow keys to get from one field to another, getting trapped in multiline text fields, and confusion when in a multiline text field of whether arrow keys will move you within that field or move you to the next field.
